### PR TITLE
GH-15967 - add xgboost custom metrics

### PR DESCRIPTION
**Pull request by valenad1**
#15967

### CODE REVIEW

File: h2o-py/tests/testdir_algos/xgboost/pyunit_xgboost_eval_metric_stopping.py [hunk 1]
```diff
@@ -1,9 +1,15 @@
 import sys
 
 sys.path.insert(1, "../../../")
+import h2o
```
Comment (maurever) [inline h2o-py/tests/testdir_algos/xgboost/pyunit_xgboost_eval_metric_stopping.py:4]: Importing only the upload_custom_metric function would be better than importing the whole h2o package.

**Review by maurever — COMMENTED** [review]
LGTM, just one suggestion.

File: h2o-py/tests/testdir_algos/xgboost/pyunit_xgboost_eval_metric_stopping.py (hunk 1 above)
Comment (valenad1) [reply h2o-py/tests/testdir_algos/xgboost/pyunit_xgboost_eval_metric_stopping.py]: done, thanks

**Review by maurever — APPROVED** [review]
LGTM, thanks @valenad1!

**Review by syzonyuliia — APPROVED** [review]

**Pull request merged**